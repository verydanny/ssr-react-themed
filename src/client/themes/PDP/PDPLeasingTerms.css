@value (
  Black,
  Gray1,
  Gray4
) from 'ui/shared/themes/colors/colors.css';

@value (
  HorizontalPadding
) from 'ui/shared/themes/variables/variables.css';

.PDPLeasingTerms {
  background-color: Gray4;
  display: none;
  padding: 13px HorizontalPadding;
}

.PDPLeasingTerms-expanded {
  display: flex;
  justify-content: space-between;
}

.PDPLeasingTerms_Terms {
  margin-top: 20px;
  width: 185px;
}

.PDPLeasingTerms_Terms div {
  color: Gray1;
  margin-bottom: 17px;
}

.PDPLeasingTerms_Header {
  color: Black;
  font-weight: bold;
  padding-right: 25px;
  width: 118px;
  overflow-wrap: break-word;
}
