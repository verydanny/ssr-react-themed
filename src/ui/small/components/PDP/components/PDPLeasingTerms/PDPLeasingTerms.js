import React from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed'
import cn from 'classnames'
import { handleBR } from 'lib/utils/handleBR'

import PDPSection from '../PDPSection'

const getLeasingTheme = ({ expanded, theme }) =>
  cn(theme.PDPLeasingTerms, { [theme['PDPLeasingTerms-expanded']]: expanded })

const PDPLeasingTerms = ({ terms, theme }) => {
  if (terms) {
    return (
      <PDPSection
        category="Leasing Terms"
        dataTid="pdpLeasingTerms"
        description="View our leasing terms."
        theme={theme}
        renderHidable={({ expanded }) => (
          <div className={getLeasingTheme({ expanded, theme })}>
            <h3
              className={theme.PDPLeasingTerms_Header}
              data-tid="pdpLeasingTerms_InnerHeader"
            >
              Leasing Terms
            </h3>
            <div
              className={theme.PDPLeasingTerms_Terms}
              data-tid="pdpLeasingTerms_Terms"
            >
              {handleBR(terms)}
            </div>
          </div>
        )}
      />
    )
  }

  return null
}

PDPLeasingTerms.propTypes = {
  terms: PropTypes.string,
  theme: PropTypes.object,
}

PDPLeasingTerms.defaultProps = {
  theme: {},
}

export default themed(/^PDPLeasingTerms/, { pure: true })(PDPLeasingTerms)
