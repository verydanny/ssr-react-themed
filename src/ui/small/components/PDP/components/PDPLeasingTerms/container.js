import { connect } from 'react-redux'
import { getSpecialTermsText } from 'app/selectors/pdpListing'
import PDPLeasingTerms from './PDPLeasingTerms'

const mapStateToProps = state => ({
  terms: getSpecialTermsText(state),
})

const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)(PDPLeasingTerms)

