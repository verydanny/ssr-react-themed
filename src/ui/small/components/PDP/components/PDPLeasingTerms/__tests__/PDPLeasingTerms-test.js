import React from 'react'
import { shallow } from 'enzyme'
import PDPLeasingTerms from '../PDPLeasingTerms'

const defaultProps = {
  theme: {},
}

const factory = props =>
  <PDPLeasingTerms.WrappedComponent {...defaultProps} {...props} />

describe('<PDPLeasingTerms />', () => {
  describe('with some terms', () => {
    const wrapper = shallow(factory({ terms: 'Some leasing terms' }))

    it('should render', () => {
      expect(wrapper.exists()).toBe(true)
      expect(wrapper.html()).toBeTruthy()
    })

    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('without any terms', () => {
    const wrapper = shallow(factory())

    it('should not render', () => {
      expect(wrapper.html()).toBe(null)
    })
  })
})
