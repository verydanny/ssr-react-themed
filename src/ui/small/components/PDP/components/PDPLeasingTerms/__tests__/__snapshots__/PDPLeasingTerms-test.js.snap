// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PDPLeasingTerms /> with some terms should match snapshot 1`] = `

<section data-tid="pdpLeasingTerms">
  <div role="Button"
       tabindex="0"
       data-tid="pdpLeasingTerms_Container"
       class
  >
    <div>
      <h2 data-tid="pdpLeasingTerms_Header">
        Leasing Terms
      </h2>
      <button data-tid="pdpLeasingTerms_Button"
              class
      >
      </button>
    </div>
    <div>
      <span data-tid="pdpLeasingTerms_Description">
        View our leasing terms.
      </span>
    </div>
  </div>
  <div>
    <div class>
      <h3 data-tid="pdpLeasingTerms_InnerHeader">
        Leasing Terms
      </h3>
      <div data-tid="pdpLeasingTerms_Terms">
        <div>
          Some leasing terms
        </div>
      </div>
    </div>
  </div>
</section>

`;
